src/ncp/ncp_base_mtd.cpp

```diff
@@ -4552,6 +4552,35 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_PHY_PCAP_ENABLED>(voi
     return error;
 }
 
+template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_NET_LEAVE_GRACEFULLY>(void) { return OT_ERROR_NONE; }
+
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_NET_LEAVE_GRACEFULLY>(void)
+{
+    otError error = OT_ERROR_NONE;
+    bool    leave;
+
+    SuccessOrExit(error = mDecoder.ReadBool(leave));
+
+    error = otThreadDetachGracefully(mInstance, ThreadDetachGracefullyHandler, this);
+
+exit:
+    return error;
+}
+
+void NcpBase::ThreadDetachGracefullyHandler(void *aContext)
+{
+    static_cast<NcpBase *>(aContext)->ThreadDetachGracefullyHandler();
+}
+
+void NcpBase::ThreadDetachGracefullyHandler(void)
+{
+    spinel_status_t status = SPINEL_STATUS_THREAD_DETACH_GRACEFULLY_DONE;
```

Comment:
Is SPINEL_STATUS_OK sufficient?
```suggestion
    spinel_status_t status = SPINEL_STATUS_OK;
```

Reply:
Now I have used a different approach: Emit `SPINEL_PROP_NET_DETACH_GRACEFULLY` to the host to indicate that it has been completed.